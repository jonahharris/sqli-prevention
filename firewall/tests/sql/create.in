>>> Sada testovacich SQL dotazu: create
# database
CREATE DATABASE db1;
CREATE SCHEMA db2;
CREATE OR REPLACE DATABASE db1;
CREATE DATABASE IF NOT EXISTS db1;
CREATE DATABASE IF NOT EXISTS db1 DEFAULT CHARACTER SET = 'keybcs2';
CREATE DATABASE IF NOT EXISTS db1 DEFAULT COLLATE = 'keybcs2_bin';
CREATE DATABASE IF NOT EXISTS db1 DEFAULT COLLATE 'keybcs2_bin';
CREATE DATABASE czech_slovak_names CHARACTER SET = 'keybcs2' COLLATE = 'keybcs2_bin';
# table
CREATE TABLE t1(a INT(6), b CHAR);
CREATE OR REPLACE TABLE t1(a INT(125));
CREATE TEMPORARY TABLE t1(a INT(6), c YEAR);
CREATE TABLE IF NOT EXISTS t1(a INT(6), b CHAR, c REAL);
CREATE TABLE db1.t1(a INT(6), b REAL UNSIGNED);
CREATE OR REPLACE TABLE db1.t1(a INT(6), b REAL (14) ZEROFILL);
CREATE TEMPORARY TABLE db1.t1(a FLOAT(6), b DECIMAL);
CREATE TABLE IF NOT EXISTS db1.t1(a INT(6), b BLOB);
CREATE TABLE t2 (a bigint primary key);
CREATE TABLE t2 (a bigint primary key, b int not null);
CREATE TABLE t1 like t2;
CREATE TABLE t1 (like t2);
CREATE TABLE bar (n INT(8) UNIQUE (n)) SELECT n FROM foo;
CREATE TABLE bar (n INT(8) UNIQUE (n)) AS SELECT n FROM foo;
CREATE TABLE bar (n INT(8) UNIQUE (n)) IGNORE AS SELECT n FROM foo;
CREATE TABLE bar (n INT(8) UNIQUE (n)) REPLACE AS SELECT n FROM foo;
CREATE TABLE foo (a TINYINT NOT NULL) SELECT b + 1 AS a FROM bar;
CREATE TABLE t1 ( a INT NOT NULL, PRIMARY KEY (a));
