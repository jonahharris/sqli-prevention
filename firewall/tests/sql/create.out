>>> Sada testovacich SQL dotazu: create
# database
DATABASE db1;CREATE 0 0;STMT;1;0;
DATABASE db2;CREATE 0 0;STMT;1;0;
DATABASE db1;CREATE 1 0;STMT;1;0;
DATABASE db1;CREATE 0 1;STMT;1;0;
STRING '***';CHARACTER SET 1 4;DATABASE db1;CREATE 0 1;STMT;1;0;
STRING '***';COLLATE 1 4;DATABASE db1;CREATE 0 1;STMT;1;0;
STRING '***';COLLATE 1 0;DATABASE db1;CREATE 0 1;STMT;1;0;
STRING '***';CHARACTER SET 0 4;STRING '***';COLLATE 0 4;DATABASE czech_slovak_names;CREATE 0 0;STMT;1;0;
# table
TABLE t1;COLUMN 40006 a;COLUMN 120000 b;CREATE 0 0 0 2;STMT;1;0;
TABLE t1;COLUMN 40125 a;CREATE 1 0 0 1;STMT;1;0;
TABLE t1;COLUMN 40006 a;COLUMN 100005 c;CREATE 0 1 0 2;STMT;1;0;
TABLE t1;COLUMN 40006 a;COLUMN 120000 b;COLUMN 70000 c;CREATE 0 0 1 3;STMT;1;0;
TABLE db1.t1;COLUMN 40006 a;COLUMN 71000 b;CREATE 0 0 0 2;STMT;1;0;
TABLE db1.t1;COLUMN 40006 a;COLUMN 72014 b;CREATE 1 0 0 2;STMT;1;0;
TABLE db1.t1;COLUMN 90006 a;COLUMN 110000 b;CREATE 0 1 0 2;STMT;1;0;
TABLE db1.t1;COLUMN 40006 a;COLUMN 160002 b;CREATE 0 0 1 2;STMT;1;0;
TABLE t2;ATTR PRIKEY;COLUMN 60000 a;CREATE 0 0 0 1;STMT;1;0;
TABLE t2;ATTR PRIKEY;COLUMN 60000 a;ATTR NOTNULL;COLUMN 40000 b;CREATE 0 0 0 2;STMT;1;0;
TABLE t1;TABLE t2;CREATE LIKE 0 0 0 0;STMT;1;0;
TABLE t1;TABLE t2;CREATE LIKE 0 0 0 0;STMT;1;0;
TABLE bar;COLUMN n;ATTR UNIQUEKEY 1;COLUMN 40008 n;NAME n;TABLE foo;SELECT 0 1 1;OPTSELECT 0 0;CREATE SELECT 0 0 0 1;STMT;1;0;
TABLE bar;COLUMN n;ATTR UNIQUEKEY 1;COLUMN 40008 n;NAME n;TABLE foo;SELECT 0 1 1;OPTSELECT 0 1;CREATE SELECT 0 0 0 1;STMT;1;0;
TABLE bar;COLUMN n;ATTR UNIQUEKEY 1;COLUMN 40008 n;NAME n;TABLE foo;SELECT 0 1 1;OPTSELECT 1 1;CREATE SELECT 0 0 0 1;STMT;1;0;
TABLE bar;COLUMN n;ATTR UNIQUEKEY 1;COLUMN 40008 n;NAME n;TABLE foo;SELECT 0 1 1;OPTSELECT 2 1;CREATE SELECT 0 0 0 1;STMT;1;0;
TABLE foo;ATTR NOTNULL;COLUMN 10000 a;NAME b;NUMBER 1;ADD;ALIAS a;TABLE bar;SELECT 0 1 1;OPTSELECT 0 0;CREATE SELECT 0 0 0 1;STMT;1;0;
TABLE t1;ATTR NOTNULL;COLUMN 40000 a;COLUMN a;PRIKEY 1;CREATE 0 0 0 2;STMT;1;0;